packages/block-editor/src/components/list-view/block.js

```diff
@@ -130,8 +130,8 @@ function ListViewBlock( {
 		// since that menu is part of the toolbar in the editor canvas.
 		// List View respects this by also hiding the block settings menu.
 		hasBlockSupport( blockName, '__experimentalToolbar', true ) &&
-		// Don't show the settings menu if block is disabled or content only.
-		blockEditingMode === 'default';
+		// Don't show the settings menu if block is disabled.
+		blockEditingMode !== 'disabled';
```

Comment:
I think the whole block wouldn't be shown when disabled, so the check might be unnecessary. Worth looking into it.